docs/root/configuration/network_filters/mongo_proxy_filter.rst

```diff
@@ -182,11 +182,13 @@ Dynamic Metadata
 
 The Mongo filter emits the following dynamic metadata when enabled via the
 :ref:`configuration <envoy_api_field_config.filter.network.mongo_proxy.v2.MongoProxy.emit_dynamic_metadata>`.
-This dynamic metadata is available as one struct per message under a list named *messages*.
+This dynamic metadata is available as key-value pairs where the key
+represents the database and the collection being accessed, and the value is
+a list of operations performed on the collection.
 
 .. csv-table::
   :header: Name, Type, Description
   :widths: 1, 1, 2
 
-  operation, string, The operation to be executed.
-  resource, string, The resource name in *db.collection* format on which the operation is to be executed.
+  key, string, The resource name in *db.collection* format.
+  value, array, A list of strings representing the operations executed on the resource (insert/update/query/delete).
```

Comment:
Per our offline convo, FYI, in order for this to be actually useful you are going to have to log/store query commands.